rfc/rfc-50/rfc-50.md

```diff
@@ -0,0 +1,93 @@
+
+But for Flink , Write tasks are resident services due to their pure streaming model. There is also no highly reliable communication mechanism between the user-side JM and the TM in Flink, so the TM needs to obtain the latest instant by polling the timeline for writing.
+
+![](ComparisonDiagram.png)
+
+### Current
+
+![](CurrentDesign.png)
+
+The current design implementation has two main problems with the convergence timeline
+- Since the timeline of the task is pulled from the Embedded timeline service, the refresh mechanism of the Embedded timeline service itself will doesn't work
+- MetaClient and HoodieTable are decoupled. Obtain the timeline in MetaClient and then request the Embedded timeline service to obtain file-related information through the FileSystemViewManager in HoodieTable combined with the timeline. There are circular dependencies and problems in the case of using MetaClient alone without creating HoodieTable
+
+## Implementation
+
+### Design target
+
+The goal of this solution is to converge the acquisition of timelines and obtain them through the Embedded timeline service uniformly. The timeline is pulled through HDFS only when the Embedded timeline service is not started.
+
+### Converge the request to loop instant in Flink to JM
+
+- Store the latest instant on the Embedded Timeline Server. Every time JM modifies the instant state, it actively performs a sync to Embedded Timeline Server
+- Return the latest instant directly when the task pulls the latest instant
+
+### Converge the request to pull instant in meta  client initialization to JM
+
+- Abstract the timeline-related acquisition methods into the new interface TableTimelineView, and create the corresponding TimelineViewManager in MetaClient, and obtain the timeline through TimelineViewManager.
+
+![](Design.png)
+
+### Flink optimization before and after schematic diagram
+
+![](SchematicDiagram.png)
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
+    - Since the Embedded Timeline Service is used to pull the timeline, users who use flink to write to hudi will observe that file system requests are greatly reduced, thereby reducing the pressure on the file system. 
+    - However, in a scenario with a relatively high degree of parallelism, it may be necessary to increase the resources of JM to ensure the effectiveness of the response
+- If we are changing behavior how will we phase out the older behavior?
+    - Add a configuration to control this behavior
+- If we need special migration tools, describe them here.
+    - No special migration tools will be necessary
+- When will we remove the existing behavior
+    - In subsequent releases (1.0 or later)
+## Test Plan
+
+Test plan
+No additional regression testing is required, as the behavior of MetaClient's active timeline has not been changed
```

Comment:
Understand that regression test is not applicable. I'd suggest at least note down what sort of integration test should be carried out to ensure the functional correctness. Are we adding some new functional tests in the CI or can we enable the config for existing test cases to cover the new code path?

Reply:
We can enable the config for existing test cases to cover the new code path, like change Test to ParameterizedTest.